### DISCUSSION TITLE
npm i dominos on ec2 server error w/ node-expat

### DISCUSSION THREAD

**Discussion by lastraum** [RIAEvangelist/node-dominos-pizza-api#103] [General]
Hey Brandon,

Any reason why npm i dominos won't work on my new ec2 server and receiving a node-expat error?

`npm ERR! code 1
npm ERR! path /home/ec2-user/lk/dominos/node_modules/node-expat
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! make: Entering directory `/home/ec2-user/lk/dominos/node_modules/node-expat/build'
npm ERR!   CC(target) Release/obj.target/expat/deps/libexpat/lib/xmlparse.o
npm ERR! make: Leaving directory `/home/ec2-user/lk/dominos/node_modules/node-expat/build'
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.2
npm ERR! gyp info using node@15.11.0 | linux | x64
npm ERR! gyp info find Python using Python version 2.7.18 found at "/usr/bin/python"
npm ERR! gyp info spawn /usr/bin/python
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/home/ec2-user/.nvm/versions/node/v15.11.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/ec2-user/lk/dominos/node_modules/node-expat/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/ec2-user/.nvm/versions/node/v15.11.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/ec2-user/.cache/node-gyp/15.11.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/home/ec2-user/.cache/node-gyp/15.11.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/home/ec2-user/.nvm/versions/node/v15.11.0/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/home/ec2-user/.cache/node-gyp/15.11.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/home/ec2-user/lk/dominos/node_modules/node-expat',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! make: cc: Command not found
npm ERR! make: *** [Release/obj.target/expat/deps/libexpat/lib/xmlparse.o] Error 127
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/home/ec2-user/.nvm/versions/node/v15.11.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:378:20)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
npm ERR! gyp ERR! System Linux 4.14.219-161.340.amzn2.x86_64
npm ERR! gyp ERR! command "/home/ec2-user/.nvm/versions/node/v15.11.0/bin/node" "/home/ec2-user/.nvm/versions/node/v15.11.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /home/ec2-user/lk/dominos/node_modules/node-expat
npm ERR! gyp ERR! node -v v15.11.0
npm ERR! gyp ERR! node-gyp -v v7.1.2
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in:`

**Comment by RIAEvangelist** [comment]
Looks like an issue compiling xml2json a dependency for reading the occasional soap responses. I will do some googling.

**Comment by RIAEvangelist** [comment]
OK, @lastraum can you try removing your dependency on the production `dominos` and replace it with  `dominos@3.0.1-beta.0`

That uses the latest `xml2json` and may fix the issue. LMK.

**Comment by RIAEvangelist** [comment]
I made a boo boo that was my first time doing a beta release and I forgot to tag the npm publish as beta, so I just republished as prod v3.0.2 so you can just update your version and try.

**Reply by lastraum** [reply]
what do you mean? in my package file?

"dominos": "3.0.1-beta.0"

**Reply by RIAEvangelist** [reply]
that works, but just update it to `3.0.2` as I made it prod now.

**Reply by lastraum** [reply]
ok. i'm having a really hard time following how to just get the most recent stable version installed....

npm i dominos in an empty folder brings up a ton of error on ec2...

LEADE_CASE errors with expat as well as CHECK_NAME_CASES....

**Reply by RIAEvangelist** [reply]
what version is it trying to install?

**Reply by RIAEvangelist** [reply]
nevermind, if its in an empty folder that's 3.0.2. what OS ? I assume ubuntu?

**Reply by lastraum** [reply]
linux....

i type npm i dominos@3.0.2 and still same error....been trying to figure this out all day and building a workaround simultaneously...

**Reply by RIAEvangelist** [reply]
I had assumed linux ;) was looking for more error, but I am getting a feel for something here.

have you tried updating?

```js

sudo apt update
sudo aptt upgrade
sudo apt install build-essential

```

**Reply by lastraum** [reply]
done that, however the last command doesn't work

sudo apt install build-essential 

says missing destination file

**Reply by RIAEvangelist** [reply]
If that doesnt work, try installing a non edge version of node just incase.

```sh

nvm install 14.16
nvm use 14.16

```

**Reply by RIAEvangelist** [reply]
Ill google not being able to install build essential

**Reply by lastraum** [reply]
message me on discord if you can Lastraum#0153

**Reply by RIAEvangelist** [reply]
Discord says that's not a valid id. I also added an extra zero on the end to make the number valid, but still does not work. Please be careful and send the correct id for me.

**Reply by RIAEvangelist** [reply]
@lastraum I'm going to go back to work here pretty quick. Waiting on the correct discord ID

**Reply by lastraum** [reply]
lastraumonline@gmail.com 

that discord id should work

**Reply by RIAEvangelist** [reply]
@lastraum  I appreciate that you think a malformatted discord ID should work, but unfortunately, it does not. 

If you want my help setting up your Linux box, so you can use my node module, please provide the correct ID.

**Reply by lastraum** [reply]
not sure how else to get it to you sorry; i copy and paste directly from discord

**Reply by RIAEvangelist** [reply]
This is getting a bit frustrating for me. Discord has 4 numbers after the # you only copied and pasted 3 which is why I suggested you be more careful.

**Reply by lastraum** [reply]
Lastraum#0153

**Reply by lastraum** [reply]
#0153

**Reply by lastraum** [reply]
there is a 0 before the 1, it keeps deleting when i add it

**Reply by RIAEvangelist** [reply]
Thanks

**Reply by RIAEvangelist** [reply]
I would not have guessed that, I added the Zero at the end. I added you.

**Comment by RIAEvangelist** [comment]
Just for anyone reading `Amazon linux` for EC2 seems to be missing some packages required for compiling `xml2json`  so I switched to using `xml2js` which does install just fine, but has a slightly different format for tracking.

issue was resolved in [`v3.1.0`](https://github.com/RIAEvangelist/node-dominos-pizza-api/releases/tag/3.1.0)

**Comment by RIAEvangelist** [comment]
I'm adding you to the contributor section of the package.json The tracking info we got on the discord pair coding was useful.

**Comment by RIAEvangelist** [comment]
Just to confirm with everyone, this question was fixed with @lastraum when he realized he did not have a typescript types file setup as this module is pure JS.
